### PR TITLE
Make numbers more consistent

### LINKED ISSUE #1245: Consistent customisable way to use decimals
In the CSS support there's a couple of commands that recognise decimals:

https://github.com/talonhub/community/blob/d9c6373661eb937fc6fc1a29df4bfdce54b6914a/lang/css/css.talon#L27-L29

This only recognises how decimals are typically taught to be read in English-speaking classrooms (e.g. `three point one four` for `3.14`). Some people might like to be able to dictate the fractional part of decimals like whole numbers though, e.g. `twelve point thirty four` for `12.34`. Others might prefer to say `dot` instead of `point`.

We should add a `decimal_string` capture to allow this to be reused in other commands and customised globally.

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Makes numbers more consistent between prose formatters and the global number command. 

Increase consistency and additional delimiters for different locales

Fixes #1245

### CODE REVIEW

File: core/numbers/numbers.py [hunk 1]
```diff
@@ -243,30 +243,58 @@ def number_string(m) -> str:
...
+    return f"-{number}" if (m[0] in ["negative", "minus"]) else number
+
+
+@ctx.capture("number_signed", rule="<user.number_signed_string>")
+def number_signed(m) -> int:
+    """Parses a (possibly negative) number phrase, returning that number as a integer."""
+    return int(m.number_signed_string)
+
+
+@mod.capture(rule="<user.number_string> ((dot | point) <number_string>)+")
```
Comment (knausj85) [inline core/numbers/numbers.py:265]:
```suggestion
@mod.capture(rule="<user.number_string> ((dot | point) <user.number_string>)+")
```

File: core/numbers/numbers.py [hunk 2]
```diff
@@ -243,30 +243,58 @@ def number_string(m) -> str:
...
+def number_prose_with_dot(m) -> str:
+    return ".".join(m.number_string_list)
+
+
+@mod.capture(rule="<user.number_string> (comma <number_string>)+")
```
Comment (knausj85) [inline core/numbers/numbers.py:270]:
```suggestion
@mod.capture(rule="<user.number_string> (comma <user.number_string>)+")
```

File: core/numbers/numbers.py [hunk 3]
```diff
@@ -243,30 +243,58 @@ def number_string(m) -> str:
...
 def number(m) -> int:
     """Parses a number phrase, returning it as an integer."""
     return int(m.number_string)
 
 
-@ctx.capture("number_signed", rule=f"[negative|minus] <number>")
-def number_signed(m):
-    number = m[-1]
-    return -number if (m[0] in ["negative", "minus"]) else number
+@mod.capture(rule="[negative | minus] <user_number_string>")
```
Comment (knausj85) [inline core/numbers/numbers.py:252]:
```suggestion
@mod.capture(rule="[negative | minus] <user.number_string>")
```

**Review by knausj85 — CHANGES_REQUESTED** [review]
Few typos I think, otherwise I think this change makes sense

File: core/numbers/numbers.py [hunk 4]
```diff
@@ -243,30 +243,58 @@ def number_string(m) -> str:
...
+def number_signed_string(m) -> str:
+    """Parses a (possibly negative) number phrase, returning that number as a string."""
+    number = m.user_number_string
```
Comment (knausj85) [inline core/numbers/numbers.py:255]:
```suggestion
    number = m.number_string
```

**Comment by knausj85** [comment]
There's one other inconsistency that might be worth addressing while in here, or as a separate issue: 
can't dictate negative numbers with decimals.

**Comment by AndreasArvidsson** [comment]
Fixes applied.

Negative decimal numbers for something I thought about but since we have a specific signed numbers for integers do we need something similar for decimals or should we just make one capture supporting both? I think we should leave that as a separate issue.

**Review by knausj85 — APPROVED** [review]
looks good, thanks. I agree with handling the negative numbers in a separate issue.

**Comment by AndreasArvidsson** [comment]
Issue filed
https://github.com/talonhub/community/issues/1615

**Pull request merged**